examples/sum-types/F.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module F (fCategorified) where
+
+import qualified Categorifier.C.CExpr.Cat as C
+import Categorifier.C.CExpr.Cat.TargetOb (TargetOb)
+import Categorifier.C.CTypes.CGeneric (CGeneric)
+import qualified Categorifier.C.CTypes.CGeneric as CG
+import Categorifier.C.CTypes.GArrays (GArrays)
+import Categorifier.C.KTypes.C (C)
+import Categorifier.C.KTypes.KEnum (KEnum, toKEnum)
+import Categorifier.C.KTypes.Sum.Maybe (KMaybe, kJust, kNothing)
+import qualified Categorifier.Categorify as Categorify
+import Categorifier.Client (deriveHasRep)
+import qualified Categorifier.Client
+import Data.Int (Int32)
+import Data.Word (Word64)
+import GHC.Generics (Generic)
+
+data Pet = Cat | Dog | Capybara
+  deriving (Bounded, Enum, Eq, Ord, Generic)
+
+deriveHasRep ''Pet
+
+instance CGeneric Pet
+
+instance GArrays C Pet
+
+-- The 'TargetOb' instance for an enum type should be defined via
+-- `Categorifier.Client.Rep`, rather than `CG.Rep`.
+type instance TargetOb Pet = TargetOb (Categorifier.Client.Rep Pet)
+
+data Input = Input
+  { iInt32 :: C Int32,
+    -- | To use an enum type, wrap it with `KEnum`.
+    iEnum :: KEnum C Pet
```

Comment:
99% of the time, enums can just be used directly, without `KEnum`. I feel like this isn't quite right, it's too broad, but we only need `KEnum` if we "use" it in an `f`-parameterized context. This example should really be just using `Pet`, not `KEnum C Pet`.

Reply:
Using `Pet` directly doesn't work here because there's no `IfCat Cat Pet`. I don't immediately know whether this instance is not possible, or we just didn't implement it (see #42), but either way, `KEnum` is currently the safer thing to use.

Reply:
Ah! You're right, it's `IfCat` that forces us to use `KEnum` in general. I knew my recollection wasn't right. We should capture that on the docs for `KEnum`, and also add a comment on this example that points out why we need `KEnum` here, but don't in general.

Reply:
Updated.